report/4.Solution/GeneratedCppCode.tex

```diff
@@ -0,0 +1,86 @@
+% !TeX root = ../main.tex
+
+\subsection{Generated C++ code}
+While the previous section is primarily focused on the code generation from an abstract point of view, this section focuses on the generated C++ code.
+Since C++ is the output language of the HCL compiler, all the language concepts of HCL have to be represented in valid C++ code.
+
+\textbf{Primitive types}
+The primitive types of HCL includes numbers, text and booleans. 
+Booleans are straightforward, as HCL's boolean behaves exactly like the C++ boolean type.
+This means that the regular C++ boolean type can be used as the output type of HCL booleans.
+
+As HCL aims for simplicity, only one number type is introduced. 
+This type needs to be able to store both integers and non-integers. 
+The simple solution is to turn all HCL numbers into C++'s double type.
+The solution with doubles is not ideal, but it will fulfill the requirements of the number type in HCL.
```

Comment:
Cahnge structure of sentence and remove redundant text -> "This fulfills the requirements of the number type in HCL, even though it was not the most ideal solution"

Reply:
"...It is not the ideal solution"

Reply:
Not to be defensive, but I don't really understand the requested change.
First of all IMO both are perfectly valid English, with the exact same meaning.
Second of all the original sentence is shorter than the proposed sentence, so the part about removing redundant stuff seems strange.
Again, not being to defensive here, just genuinely puzzled why the proposed sentence is better? :smile:

Reply:
Just fyi this one is the only open comment as far as I can see.

Reply:
Yea, that's a good point haha

Reply:
that is a good point haha